buzzmobile/process/bearing/bearing.py

```diff
@@ -40,13 +49,25 @@ def bearing(fix):
         lon1 = math.radians(bearings['last_fix'].longitude)
         lat2 = math.radians(fix.latitude)
         lon2 = math.radians(fix.longitude)
-        y = math.sin(lon2 - lon1) * math.cos(lat2)
-        x = math.cos(lat1)*math.sin(lat2) - math.sin(lat1)*math.cos(lat2)*math.cos(lon2 - lon1)
-        angle = math.atan2(y, x)
-        computedBearing = (angle + 2*math.pi) % 2*math.pi
-        bearings['med_filter'].add(computedBearing)
 
-        bearing_pub.publish(bearings['med_filter'].median())
+        #Calculates great-circle distance between positions
+        a = (math.pow(math.sin(lat2 - lat1), 2)
+             + math.cos(lat1) * math.cos(lat2)
+             * math.pow(math.sin((lon2 - lon1) / 2), 2))
+        c = 2 * math.atan2(math.sqrt(a), math.sqrt(1 - a))
+        distance = EARTH_RADIUS * c
+
+        if distance >= MIN_FIX_DISTANCE:
```

Comment:
I think there's this idea of a good bearing (meets the delta requirement) and a bad bearing (does not meet delta requirement)

Use last fix to update either good or bad bearing.

If there is a last good bearing,  publish that (unless its way too old, but this time part can be implemented when we're fixing #78 )

If there isnt a good bearing, publish the last bearing we were able to get (which can be a bad bearing)

Idk how exactly that translates into implementation, but we want to prioritize good bearings, but not completely fail if we don't have a good bearing. I also don't like the idea of defaulting to 0 rad bearing because our fixes, albeit noisy, might still provide a better bearing than a hardcoded 0 rad.